### DISCUSSION TITLE
custom loss which depends on input features

### DISCUSSION THREAD

**Discussion by chanansh** [catboost/catboost#2874] [Q&A]
I need to compute loss which is based on the input features as well, e.g. loss = foo (approx, target, features)

however, in the docs I see this example:

```
class MultiRmseObjective(MultiTargetCustomObjective):
    def calc_ders_multi(self, approx, target, weight):
        assert len(target) == len(approx)
        
        w = weight if weight is not None else 1.0
        der1 = [(target[i] - approx[i]) * w for i in range(len(approx))]
        der2 = [-w for i in range(len(approx))]

        return (der1, der2)

```

so that `calc_ders_multi` does not get as input the features. 

Is my usecase possible?

**Comment by andrey-khropov** [comment]
> Is my usecase possible?
 
 Not at the moment. There is an issue for such functionality: #1601
